recipes/build_recipe_macos.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# -----------------------------------------------------------------------------
+# Run this from the root of the OpenTurbine repository with the following arguments:
+# ./recipes/build_recipe_macos.sh <path_to_openturbine_root> <path_to_spack_root>
+# -----------------------------------------------------------------------------
+
+# Exit on error
+set -e
+
+# Clone OpenTurbine repository if not present in the provided path
+openturbine_path=$(dirname "$1") # Get the directory of the provided path
+if [ ! -d "$openturbine_path/openturbine" ]; then
+    git clone --recursive https://github.com/Exawind/openturbine.git $openturbine_path/openturbine
+fi
+cd $openturbine_path/openturbine
+
+# Install Spack if not present in the provided path
+spack_path=$(dirname "$2") # Get the directory of the provided path
+if [ ! -d "$spack_path/spack" ]; then
+    git clone https://github.com/spack/spack.git $spack_path/spack
+fi
+source $spack_path/spack/share/spack/setup-env.sh
+spack compiler find
+spack external find
+
+# Install required packages using Spack if not present
+install_if_missing() {
+    if ! spack find -l "$1" | grep -q "$1"; then
+        echo "Installing $1..."
+        spack install "$1"
+    else
+        echo "$1 is already installed."
+    fi
+}
+
+install_if_missing googletest
+install_if_missing llvm
+install_if_missing cppcheck
+install_if_missing yaml-cpp
+install_if_missing "vtk~mpi~opengl2"
+install_if_missing "trilinos@16.0.0~mpi~epetra"
+
+# Load required packages
+spack load llvm
+spack load cppcheck
+spack load trilinos
+spack load googletest
+spack load yaml-cpp
+spack load vtk
+
+# Find gfortran compiler
+if ! command -v brew &> /dev/null; then
+    echo "Homebrew is not installed. Please install it first."
+    exit 1
+fi
+export FC=$(find /opt/homebrew/ -name gfortran | tr ' ' '\n' | grep "/gcc/.*gfortran")
```

Comment:
I think we should just check that the user has defined FC in their environment and tell them to do so if they have not.  They may have a preferred Fortan compiler that they want to use.

Reply:
Sounds good to me